### PR TITLE
Chore: beautify

### PR DESCRIPTION
**Pull request by pr0n00gler**
Just some fixes:
* excessive ValidateMsg() are removed;
* proper tests for ValidateMsg() are added;
* proper errors in ICTX;
* add a public model for TransactionsFilter to be used in validation on Neutron's side and reused by relayer;
* add thorough validation for transactions filter on register interchain query.

### CODE REVIEW

**Comment by foxpy** [comment]
@pr0n00gler please rebase this branch on `main`

File: wasmbinding/test/custom_query_test.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ func (suite *CustomQuerierTestSuite) TestInterchainQueryResultNotFound() {
 	}
 	resp := icqtypes.QueryRegisteredQueryResultResponse{}
 	err := suite.queryCustom(ctx, contractAddress, query, &resp)
-	expectedErrMsg := "Generic error: Querier contract error: codespace: interchainqueries, code: 1115: query wasm contract failed"
+	expectedErrMsg := "Generic error: Querier contract error: codespace: interchainqueries, code: 1116: query wasm contract failed"
```
Comment (foxpy) [inline wasmbinding/test/custom_query_test.go]:
```suggestion
	expectedErrMsg := fmt.Sprintf("Generic error: Querier contract error: codespace: interchainqueries, code: %d: query wasm contract failed", icqtypes.ErrNoQueryResult.ABCICode())
```
what do you think? we wouldn't have to update this part of code anymore

File: x/interchainqueries/types/types.go [hunk 2]
```diff
@@ -125,3 +114,66 @@ func (keys KVKeys) String() string {
...
+			}
+		}
+		if f.Field == "" {
+			return fmt.Errorf("transactions filter condition idx=%d is invalid: field couldn't be empty", idx)
+		}
+		switch value := f.Value.(type) {
+		case string:
+		case float32:
+			// despite json turns numbers into float, decimals are not allowed by tendermint API
+			if value != float32(int32(value)) {
```
Comment (foxpy) [inline x/interchainqueries/types/types.go]: is float32 even in JSON standard? I thought only float64 is used

File: x/interchaintxs/types/errors.go [hunk 3]
```diff
@@ -9,5 +9,8 @@ var (
 	ErrInvalidICAOwner           = sdkerrors.Register(ModuleName, 1100, "invalid interchain account interchainAccountID")
 	ErrInvalidAccountAddress     = sdkerrors.Register(ModuleName, 1101, "invalid account address")
 	ErrInterchainAccountNotFound = sdkerrors.Register(ModuleName, 1102, "interchain account not found")
-	ErrNotContract               = sdkerrors.Register(ModuleName, 1113, "not a contract")
+	ErrNotContract               = sdkerrors.Register(ModuleName, 1103, "not a contract")
```
Comment (foxpy) [inline x/interchaintxs/types/errors.go:12]: image
you sure that's ok?

**Review by foxpy — DISMISSED** [review]
Looks good by my standards

File: x/interchaintxs/types/errors.go (hunk 3 above)
Comment (mmikev) [reply x/interchaintxs/types/errors.go:12]: Yes, different modules can have the same code errors, because uniqueness is _(module, code)_ combo

**Review by foxpy — DISMISSED** [review]

**Review by foxpy — DISMISSED** [review]

**Review by zavgorodnii — DISMISSED** [review]

File: x/interchainqueries/keeper/msg_server.go [hunk 4]
```diff
@@ -43,11 +43,6 @@ func (k msgServer) RegisterInterchainQuery(goCtx context.Context, msg *types.Msg
 		return nil, sdkerrors.Wrapf(types.ErrNotContract, "%s is not a contract address", msg.Sender)
 	}
 
-	if err := msg.ValidateBasic(); err != nil {
```
Comment (swelf19) [inline x/interchainqueries/keeper/msg_server.go:46]: why have you removed  this one validate, but left in `RemoveInterchainQuery` `UpdateInterchainQuery` methods

**Review by foxpy — DISMISSED** [review]

**Review by swelf19 — CHANGES_REQUESTED** [review]
please remove all excessive validates

File: x/interchainqueries/keeper/msg_server.go (hunk 4 above)
Comment (mmikev) [reply x/interchainqueries/keeper/msg_server.go:46]: Done

**Review by sotnikov-s — APPROVED** [review]

**Review by swelf19 — APPROVED** [review]

**Pull request merged**